src/HttpMethodsClient.php

```diff
@@ -11,6 +11,14 @@
  *
  * Use this interface when you do not have PSR-7 RequestInterface instances available.
  *
+ * Implementations should be configured in their constructor. For runtime behaviour changes, there
+ * is an $options parameter. It SHOULD be validated by the client and an exception thrown if
+ * unknown options are passed.
```

Comment:
When we started this project we had a ConfigurableRequest, which wrapped the original request and had a `getOptions` method.

To be honest: how often do you need different timeout option in a client?